activerecord/lib/active_record/nested_attributes.rb

```diff
@@ -398,7 +398,7 @@ def assign_nested_attributes_for_one_to_one_association(association_name, attrib
         if attributes.respond_to?(:permitted?)
           attributes = attributes.to_h
         end
-        attributes = attributes.with_indifferent_access
+        attributes = (attributes || {}).with_indifferent_access
```

Comment:
Could also do `attributes = attributes&.with_indifferent_access || { }`. Or maybe we should raise an exception?